paddle/fluid/pybind/eager_math_op_patch.cc

```diff
@@ -220,6 +220,10 @@ static PyObject* tensor__add__method(TensorObject* self,
         self_tensor = cast_ad_func(self_tensor, DataType::FLOAT32);
       }
     } else if (PyCheckInteger(other_obj) || IsNumpyType(other_obj)) {
+      if (PyCheckInteger(other_obj) && self_tensor.dtype() == DataType::BOOL) {
+        eager_gil_scoped_release guard;
+        self_tensor = cast_ad_func(self_tensor, DataType::INT64);
+      }
       other = CastPyArg2Double(other_obj, "__add__", 0);
```

Comment:
if dtype of self_tensor is int dtype, other should not be cast to double?

Reply:
scalar_ad_func need float input, but it don't need to cast to double. I will fix it.